src/main/java/com/example/onlinebookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -11,19 +11,19 @@
 @RequiredArgsConstructor
 @Component
 public class BookSpecificationBuilder implements SpecificationBuilder<Book, BookSearchParameters> {
-    private SpecificationProviderManager<Book> bookSpecificationProviderManager;
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
 
     @Override
     public Specification<Book> build(BookSearchParameters searchParams) {
         Specification<Book> specification = Specification.where(null);
         if (isParameterPresent(searchParams.author())) {
-            addSpecification(specification, "author", searchParams.author());
+            specification = addSpecification(specification, "author", searchParams.author());
```

Comment:
Is it possible to return specification in if statements body? You will avoid creating null specification

Reply:
I can, but then i will return 1 specification instead of several or there is another approach?